subprojects/docs/src/docs/userguide/antlr_plugin.adoc

```diff
@@ -13,58 +13,39 @@
 // limitations under the License.
 
 [[antlr_plugin]]
-== The ANTLR Plugin
+= The ANTLR Plugin
 
 The ANTLR plugin extends the Java plugin to add support for generating parsers using http://www.antlr.org/[ANTLR].
 
-[NOTE]
-====
-
-The ANTLR plugin supports ANTLR version 2, 3 and 4.
-
-====
-
+NOTE: The ANTLR plugin supports ANTLR version 2, 3 and 4.
 
 [[sec:antlr_usage]]
-=== Usage
+== Usage
 
 To use the ANTLR plugin, include the following in your build script:
 
-++++
-<sample id="useAntlrPlugin" dir="antlr" title="Using the ANTLR plugin">
-            <sourcefile file="build.gradle" snippet="use-plugin"/>
-        </sample>
-++++
+=== Example: Using the ANTLR plugin
+
+[source.multi-language-sample,groovy]
+.build.gradle
+----
+include::{samplesPath}/antlr/build.gradle[tag=use-plugin]
+----
 
 
 [[sec:antlr_tasks]]
-=== Tasks
+== Tasks
 
 The ANTLR plugin adds a number of tasks to your project, as shown below.
 
-.ANTLR plugin - tasks
-[cols="a,a,a,a", options="header"]
-|===
-| Task name
-| Depends on
-| Type
-| Description
+`generateGrammarSource` — type: link:{groovyDslPath}/org.gradle.api.plugins.antlr.AntlrTask.html[AntlrTask]::
```

Comment:
the " -type: ..." rendering looks kind of odd to me ("type" is bolded, "-" followed closely by ":" feels weird).  Maybe this should be something like "generateGrammarSource (type: _link_)"

Reply:
Funny you bring this up. @big-guy did, too. The format used to be exactly as you suggest, but after discussing with @pledbrook we changed it. 

I'd like to change these all at once and submit a separate PR for whatever we settle on. OK?

Reply:
@big-guy @ghale Compare to the [list of tasks in the _Java Plugin_ chapter](https://docs.gradle.org/current/userguide/java_plugin.html#sec:java_tasks). The native Asciidoc styling does look odd and makes the extra information more pronounced than the name of the task, which was never my intention.

The problems I see with the parentheses approach are:

 * It's cramped (although a space before the parenthesis might help) and obfuscates the name of the task, which is the important bit. I want the name to stand out so it's as easy as possible to pick up when scanning the page.
 * It looks like code, but it's not. We also need an approach that works for lifecycle tasks and task rules as in the [_Base Plugin chapter](https://docs.gradle.org/current/userguide/base_plugin.html#sec:base_tasks).

I think it would look much better if "type" didn't look larger than the monospace text either side. There's also not much space between the "type:" and the class name. It looks a bit better in the _Base Plugin_ chapter in which the class name is not a link (although it probably should be).

Anyway, happy for feedback on this, but I want to make sure that any changes:

 * Ensure that the prime term in the definition list is the one that stands out and is easy to pick out when scanning the page
 * Are spaced for easy reading
 * Are consistent across all the possible task list headings, such as for lifecycle tasks and task rules

Reply:
My suggestion to @eriwen was to remove the word type: completely. I think a user can make the connection from "this is the name of a task" and "this is the DSL for its type" without that word. It doesn't look like code.

It would look like this:
![image](https://user-images.githubusercontent.com/5387972/43010098-75fea0da-8c0d-11e8-92f7-5c11b527d85e.png)

For lifecycle and task rules, I have a couple of alternatives:
1. Remove the distinction and just label them with their types (`Task`, `Delete`, `Upload` or whatever).  If we want to highlight that they're lifecycle tasks or generated by task rules, we could add that to the description and link to another user guide section about lifecycle tasks or task rules.
1. Keep the distinction, but link to the user guide section that explains what they are like we do for task types.

I would lean towards the first option because I don't think "task rules" are special enough to get their own category and we lose information (the task type).  Lifecycle tasks are a bigger concept, so I could see some extra information about that.
It would look like this:
![image](https://user-images.githubusercontent.com/5387972/43010746-4491fde2-8c0f-11e8-9fbe-48917d4305ad.png)

Reply:
I think either making the "type:" text less prominent (i.e. smaller font, maybe not bold) or removing it altogether (sterling's suggestion) would be improvement over how it looks now.

Reply:
I think we could remove "type:". My hope is that the dotted underline and monospace font make it quite clear that this is a type and that it's linking to a definition or reference for that type. 

Do you agree @pledbrook?

Reply:
Although the change is much neater, I think we should be wary of making assumptions about how easy it is to interpret the text correctly. New users that haven't encountered task types yet might be confused without some cue that helps them.

I do wonder whether it makes sense to have a legend or something similar to provide such a cue. What about:

> ## Tasks
> _(Name — type)_
> ...

The definition lists would then be tidier and more readable. The "type" could also link to a section on task types. This is one area where tables have the advantage as you can simply put "Task type" in a column header :smile:

As for the suggestion to move "lifecycle task" and "task rule" to the description, I see the merit of moving them out of the first line, but I do think these are important attributes that should stand out when someone is scanning the list.

An option that springs to mind is to treat them like "Depends on:":

> _Depends on_: `check`, `assemble`
> [_Lifecycle task_](https://docs.gradle.org/current/userguide/more_about_tasks.html#sec:lifecycle_tasks)
>
> ...

One final thing that I think I've asked about elsewhere and is somewhat tangential: is there a policy to distinguish between _lifecycle tasks_ and tasks with only task dependencies and no actions? I like to think of the latter as a broader category that I think of as _aggregate tasks_. They can encompass convenience tasks that simply combine multiple other tasks together without have the connotations of being part of the build lifecycle.

This isn't a distinction I've seen in the user manual, but it's a distinction I favour, even if it's just a semantic difference rather than a technical one.

Reply:
I like the suggestions for a legend and marking lifecycle tasks in "Depends on" with a link. 

We do want to distinguish between lifecycle tasks and aggregate tasks. I think in general we want users to get in the habit of invoking lifecycle tasks and not more specific tasks. The benefit here is that users know how to operate any given Gradle build even if they're less familiar with the project.